src/client/eppo-client.ts

```diff
@@ -517,3 +495,32 @@ export default class EppoClient implements IEppoClient {
     }
   }
 }
+
+export function checkTypeMatch(expectedType?: VariationType, actualType?: VariationType): boolean {
+  return expectedType === undefined || actualType === expectedType;
+}
+
+export function checkValueTypeMatch(
+  expectedType: VariationType | undefined,
+  value: ValueType,
+): boolean {
+  if (expectedType == undefined) {
```

Comment:
Note: This is one of the few places we want the loose equality check (`==`) as it will check `undefined` and `null`, which is what I _think_ we want (but double check me).